drivers/sensor/sbs_gauge/sbs_gauge.c

```diff
@@ -0,0 +1,266 @@
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_AVG_CURRENT:
+		status = sbs_cmd_reg_read(dev, SBS_GAUGE_CMD_AVG_CURRENT, &data->avg_current);
+		if (status < 0) {
+			LOG_ERR("Failed to read average current ");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_TEMP:
+		status = sbs_cmd_reg_read(dev, SBS_GAUGE_CMD_TEMP, &data->internal_temperature);
+		if (status < 0) {
+			LOG_ERR("Failed to read internal temperature");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_STATE_OF_CHARGE:
+		status = sbs_cmd_reg_read(dev, SBS_GAUGE_CMD_ASOC, &data->state_of_charge);
+		if (status < 0) {
+			LOG_ERR("Failed to read state of charge");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_FULL_CHARGE_CAPACITY:
+		status = sbs_cmd_reg_read(dev,
+			SBS_GAUGE_CMD_FULL_CAPACITY, &data->full_charge_capacity);
+		if (status < 0) {
+			LOG_ERR("Failed to read full charge capacity");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_REMAINING_CHARGE_CAPACITY:
+		status = sbs_cmd_reg_read(dev,
+			SBS_GAUGE_CMD_REM_CAPACITY, &data->remaining_charge_capacity);
+		if (status < 0) {
+			LOG_ERR("Failed to read remaining charge capacity");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_NOM_AVAIL_CAPACITY:
+		status = sbs_cmd_reg_read(dev,
+			SBS_GAUGE_CMD_NOM_CAPACITY, &data->nom_avail_capacity);
+		if (status < 0) {
+			LOG_ERR("Failed to read nominal available capacity");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_FULL_AVAIL_CAPACITY:
+		status = sbs_cmd_reg_read(dev,
+			SBS_GAUGE_CMD_FULL_CAPACITY, &data->full_avail_capacity);
+		if (status < 0) {
+			LOG_ERR("Failed to read full available capacity");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_TIME_TO_EMPTY:
+		status = sbs_cmd_reg_read(dev, SBS_GAUGE_CMD_AVG_TIME2EMPTY, &data->time_to_empty);
+
+		data->time_to_empty = (data->time_to_empty) & 0x00FF;
+
+		if (status < 0) {
+			LOG_ERR("Failed to read state of health");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_TIME_TO_FULL:
+		status = sbs_cmd_reg_read(dev, SBS_GAUGE_CMD_AVG_TIME2FULL, &data->time_to_full);
+
+		data->time_to_full = (data->time_to_full) & 0x00FF;
+
+		if (status < 0) {
+			LOG_ERR("Failed to read state of health");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_CYCLE_COUNT:
+		status = sbs_cmd_reg_read(dev, SBS_GAUGE_CMD_CYCLE_COUNT, &data->cycle_count);
+
+		data->cycle_count = (data->cycle_count) & 0x00FF;
+
+		if (status < 0) {
+			LOG_ERR("Failed to read state of health");
+			return -EIO;
+		}
+		break;
+
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief initialize the fuel gauge
+ *
+ * @return 0 for success
+ */
+static int sbs_gauge_init(const struct device *dev)
+{
+	struct sbs_gauge_data *const data = (struct sbs_gauge_data *const)dev->data;
+	const struct sbs_gauge_config *const cfg = (struct sbs_gauge_config *const)dev->config;
+
+	data->i2c = device_get_binding(cfg->i2c_dev);
```

Comment:
No need to get the binding. Just use the dev from the cfg and check if the device is ready

Reply:
dev used from cfg and device_is_ready used